manim/utils/updaters.py

```diff
@@ -0,0 +1,218 @@
+from __future__ import annotations
+
+import inspect
+from collections.abc import Callable
+from typing import TYPE_CHECKING, TypeVar, Union
+
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
+    from manim.mobject.mobject import Mobject
+    from manim.mobject.opengl.opengl_mobject import OpenGLMobject
+    from manim.opengl.shader import Object3D
+
+
+__all__ = [
+    "MobjectTimeBasedUpdater",
+    "MobjectNonTimeBasedUpdater",
+    "MobjectUpdater",
+    "MeshTimeBasedUpdater",
+    "MeshNonTimeBasedUpdater",
+    "MeshUpdater",
+    "SceneUpdater",
+    "MobjectUpdaterWrapper",
+    "MeshUpdaterWrapper",
+]
+
+
+M = TypeVar("M", bound=Union["Mobject", "OpenGLMobject"])
+
+MobjectTimeBasedUpdater: TypeAlias = Callable[[M, float], M]
+MobjectNonTimeBasedUpdater: TypeAlias = Callable[[M], M]
+MobjectUpdater: TypeAlias = Union[MobjectNonTimeBasedUpdater, MobjectTimeBasedUpdater]
+
+MeshTimeBasedUpdater: TypeAlias = Callable[["Object3D", float], "Object3D"]
+MeshNonTimeBasedUpdater: TypeAlias = Callable[["Object3D"], "Object3D"]
+MeshUpdater: TypeAlias = Union[MeshNonTimeBasedUpdater, MeshTimeBasedUpdater]
+
+SceneUpdater: TypeAlias = Callable[[float], None]
```

Comment:
So I was experimenting and I found this nice way to also allow extra parameters in the type alias...
https://mypy-play.net/?mypy=latest&python=3.9&gist=a5a42e6708727b64904d083f0cb97560

Reply:
Interesting, I'll think about it 👍